website/docs/r/function_app_flex_consumption.html.markdown

```diff
@@ -121,15 +121,15 @@ The following arguments are supported:
 
 * `storage_container_endpoint` - (Required) The backend storage container endpoint which will be used by this Function App.
 
-* `storage_authentication_type` - (Required) The authentication type which will be used to access the backend storage account for the Function App. Possible values are `storageaccountconnectionstring`, `systemassignedidentity`, and `userassignedidentity`.
+* `storage_authentication_type` - (Optional) The authentication type which will be used to access the backend storage account for the Function App. Possible values are `StorageAccountConnectionString`, `SystemAssignedIdentity`, and `UserAssignedIdentity`.
```

Comment:
In the schema this is actually `Required` so we shouldn't be changing it to `Optional` in the documentation
```suggestion
* `storage_authentication_type` - (Required) The authentication type which will be used to access the backend storage account for the Function App. Possible values are `StorageAccountConnectionString`, `SystemAssignedIdentity`, and `UserAssignedIdentity`.
```

https://github.com/hashicorp/terraform-provider-azurerm/blob/5e1cfe7fbae7a69efa48ef09b7f7c1b111d73061/internal/services/appservice/function_app_flex_consumption_resource.go#L132-L134